### ISSUE TITLE
Usage with  Kotlin-DSL

### ISSUE THREAD

**Issue by henrik242** [chrisgahlert/gradle-dcompose-plugin#25]
Some examples using Kotlin-DSL would be nice :)

**Comment by henrik242** [comment]
So this mostly works:
```
import com.chrisgahlert.gradledcomposeplugin.extension.DcomposeExtension
import com.chrisgahlert.gradledcomposeplugin.extension.DefaultService
import com.github.dockerjava.api.model.AuthConfig

buildscript {
    repositories {
        gradlePluginPortal()
        jcenter()
    }
    classpath {
        classpath("gradle.plugin.com.chrisgahlert:gradle-dcompose-plugin:0.15.3")
        classpath("com.github.docker-java:docker-java:3.1.1")
    }
}
apply(plugin = "com.chrisgahlert.gradle-dcompose-plugin")

configure<DcomposeExtension> {

    registry("https://my.example.com", closureOf<AuthConfig> {
        withUsername("myUser")
        withPassword("myPassword")
    })

    services(closureOf<NamedDomainObjectContainer<DefaultService>> {
        val postgres = register("postgres") {
            image = "postgres:10-alpine"
            exposedPorts = listOf("5432")
            portBindings = listOf("5432:5432")
        }

        val api = register("api") {
            dependsOn = listOf(postgres.get())
            baseDir = file("../api")
            memory = 512 * 1024 * 1024
            binds = listOf("${rootDir}:/app/hostfiles")
            exposedPorts = listOf("8080", "5005")
            portBindings = listOf("4004:8080", "5005:5005")
        }

        val buildApiImage by tasks.existing {
            dependsOn(":my-api:assemble", ":my-api:dockerJavaEnvironment")
        }
    })
```
Only problem is that I have to comment out the `registry()` part, since it fails with
```
> Docker command failed: class com.github.dockerjava.api.model.AuthConfig cannot be cast to class com.github.dockerjava.api.model.AuthConfig (com.github.dockerjava.api.model.AuthConfig is in unnamed module of loader java.net.URLClassLoader @3c2f61f; com.github.dockerjava.api.model.AuthConfig is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @55f5d130)
```

**Comment by chrisgahlert** [comment]
Thanks for sharing this. I'll give it a try, when I find some free time...

Regarding the `AuthConfig` problem: You are running into this issue, as the `docker-java` library is loaded in a separate ClassLoader, as it cannot be loaded into the Gradle classpath itself (reason: dependency conflicts). If you still need a workaround, I could add a "wrapper" implementation providing the most important config options...

**Comment by henrik242** [comment]
Thanks! A wrapper would solve my problems with this, I think, e.g:
```
registry("https://my.example.com", "myUser", "myPassword")
```

**Comment by chrisgahlert** [comment]
Sorry it took me so long. The wrapper syntax is now included in version `0.15.4`!

Thanks for waiting! ;)

**Comment by henrik242** [comment]
Thanks! :D
